src/v/config/tls_config.h

```diff
@@ -62,9 +62,11 @@ struct p12_container {
 using key_cert_container = std::variant<key_cert, p12_container>;
 
 inline constexpr std::string_view tlsv1_2_cipher_string
-  = "ECDHE-RSA-AES128-GCM-SHA256:AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:"
-    "AES256-GCM-SHA384:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-RSA-AES128-SHA:AES128-"
-    "SHA:AES128-CCM:ECDHE-RSA-AES256-SHA:AES256-SHA:AES256-CCM";
+  = "ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:AES128-GCM-"
+    "SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:AES256-"
+    "GCM-SHA384:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-CHACHA20-POLY1305:"
+    "ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:AES128-SHA:AES128-CCM:ECDHE-"
+    "RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:AES256-SHA:AES256-CCM";
```

Comment:
Some questions for my learning:
 * Do I understand it correctly that because we're using openssl as a statically-linked library, we know what tls cipher suites our version of openssl supports and therefore all of the new cipher suites should be available independent of the platform?
 * [Cloudflare supports 2 more ECDSA-based ciphers](https://developers.cloudflare.com/ssl/edge-certificates/additional-options/cipher-suites/supported-cipher-suites/) (although they are not considered Modern, only Compatible). Could you teach me how you chose these specific ciphers?
 * Would it make sense to make the supported cipher suites configurable (as a follow up)? It would mean that supporting a new cipher suite doesn't need a release next time, and it allows customers to tune the order and set of cipher suites to their liking depending on the hardware instructions available on the instance types, clients, etc.

Reply:
> we're using openssl as a statically-linked library

It's dynamically linked, but we package it
```
docker run --entrypoint /usr/bin/ldd -it docker.redpanda.com/redpandadata/redpanda:latest /opt/redpanda/libexec/redpanda
```

Reply:
> we know what tls cipher suites our version of openssl supports and therefore all of the new cipher suites should be available independent of the platform

The cipher string is just for TLSv1.2 and below.  I believe at the list of supported ciphers is static and nothing new will be introduced.  This string is passed to [`SSL_CTX_set_cipher_list`](https://docs.openssl.org/master/man3/SSL_CTX_set_cipher_list/).  According to the docs:

> The control string str for SSL_CTX_set_cipher_list(), SSL_set_cipher_list(), SSL_CTX_set_ciphersuites() and SSL_set_ciphersuites() should be universally usable and not depend on details of the library configuration (ciphers compiled in). Thus no syntax checking takes place. Items that are not recognized, because the corresponding ciphers are not compiled in or because they are mistyped, are simply ignored. Failure is only flagged if no ciphers could be collected at all.

> Could you teach me how you chose these specific ciphers?

See my answer to @BenPope below, I'll go into details on exactly this

> Would it make sense to make the supported cipher suites configurable (as a follow up)

Already ahead of you: [CORE-8272](https://redpandadata.atlassian.net/browse/CORE-8272)